lib/cib/cib_client.c

```diff
@@ -387,6 +387,30 @@ cib_client_erase(cib_t * cib, xmlNode ** output_data, int call_options)
                            output_data, call_options, NULL);
 }
 
+static int
+cib_client_init_transaction(cib_t *cib)
+{
+    op_common(cib);
+    return cib_internal_op(cib, PCMK__CIB_REQUEST_INIT_TRANSACT, NULL, NULL,
+                           NULL, NULL, cib_none, NULL);
+}
+
+static int
+cib_client_commit_transaction(cib_t *cib, int call_options)
+{
+    op_common(cib);
+    return cib_internal_op(cib, PCMK__CIB_REQUEST_COMMIT_TRANSACT, NULL, NULL,
+                           NULL, NULL, call_options, NULL);
+}
+
+static int
+cib_client_discard_transaction(cib_t *cib)
+{
+    op_common(cib);
+    return cib_internal_op(cib, PCMK__CIB_REQUEST_DISCARD_TRANSACT, NULL, NULL,
+                           NULL, NULL, cib_none, NULL);
```

Comment:
Separate commit/discard REQUESTs are fine, but just thinking out loud, an alternative would be a single end-transaction request with a boolean attribute for whether to commit.

Reply:
There's no input XML otherwise. So from a skim, we'd need to do one of:
* have separate commit/discard requests
* create some input XML just for the attribute
* add a new call option that only applies to the "end transaction" request

On the processing side, it seems reasonable to have separate functions for `based_discard_transaction()` and `based_commit_transaction()`, for clarity and to avoid an extra level of nesting.